### ISSUE TITLE
Add option to remove CDC ACM interface

### ISSUE THREAD

**Issue by gdsports** [adafruit/Adafruit_TinyUSB_ArduinoCore#19]
I propose adding a compile time option to remove the CDC ACM port. This allows
Adafruit boards to more closely emulate real USB keyboards, joysticks, etc. The
IAC class descriptor and CDC ACM descriptors confuse some embedded devices such
as KVM switches, old computer BIOS, game consoles, etc.

The PluggableUSB stack has this option. It requires some code patching but
it does work. The symbol is CDC_ENABLED.

I propose using CDC_DISABLED so when it is not defined (the default), the
behavior is unchanged. The symbol can be defined by changing the boards.txt

`adafruit_qtpy_m0.menu.usbstack.tinyusb.build.flags.usbstack=-DUSE_TINYUSB -DCDC_DISABLED`

I use arduino-cli so it is possible to define the symbol without changing
boards.txt. The VID, PID, and other USB parameters can be passed via command
line options so exact emulation is much easier.

The following change applies to this repo.

```
diff --git a/Adafruit_USBD_Device.cpp b/Adafruit_USBD_Device.cpp
index 1f51925..458c726 100644
--- a/Adafruit_USBD_Device.cpp
+++ b/Adafruit_USBD_Device.cpp
@@ -60,11 +60,17 @@ Adafruit_USBD_Device::Adafruit_USBD_Device(void)
     .bDescriptorType    = TUSB_DESC_DEVICE,
     .bcdUSB             = 0x0200,
 
+#ifdef CDC_DISABLED
+    .bDeviceClass       = 0,
+    .bDeviceSubClass    = 0,
+    .bDeviceProtocol    = 0,
+#else
     // Use Interface Association Descriptor (IAD) for CDC
     // As required by USB Specs IAD's subclass must be common class (2) and protocol must be IAD (1)
     .bDeviceClass       = TUSB_CLASS_MISC,
     .bDeviceSubClass    = MISC_SUBCLASS_COMMON,
     .bDeviceProtocol    = MISC_PROTOCOL_IAD,
+#endif
 
     .bMaxPacketSize0    = CFG_TUD_ENDPOINT0_SIZE,
```

Changes are required to the SAMD and nrf52 cores to conditionalize the call to
addInterface based on !CDC_DISABLED.

https://github.com/adafruit/ArduinoCore-samd/blob/master/cores/arduino/TinyUSB/Adafruit_TinyUSB_SAMD.cpp#L81

https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/cores/nRF5/TinyUSB/Adafruit_TinyUSB_nRF.cpp#L94

I do not have an nRF52840 board around at the moment but have tested the changes
out on an ItsyBitsy M0.

**Comment by hathach** [comment]
I prefer to do it with an extra API than using compiler switch. Though user will have to manually enter bootloader since the touch1200 won't work without CDC.

**Comment by gdsports** [comment]
Yes, removing the CDC interface means auto upload does not work so the dev must
double click the reset button to upload.

**Comment by hathach** [comment]
I am currently busy with other works, if you could, please submit an PR with additional API `Serial.disable()` to Adafruit_USBD_CDC https://github.com/adafruit/Adafruit_TinyUSB_ArduinoCore/blob/master/Adafruit_USBD_CDC.h#L31

**Comment by amowry** [comment]
I would also love to see this happen, if possible (I don't know hoe to do it myself). Thanks!

**Comment by hathach** [comment]
This is still WIP, but in the future this repo will be archived. All the sources will be moved to the Adafruit_TinyUSB_Arduino repo. The core will only has bare minimum header (CDC) for Serial usage. This feature is implemented but not released yet, I would love to hear from your feedback. You need to use developing branch
- https://github.com/adafruit/Adafruit_TinyUSB_Arduino/tree/major-v1
- for SAMD https://github.com/adafruit/ArduinoCore-samd/tree/rework-tinyusb
- for nRF https://github.com/adafruit/Adafruit_nRF52_Arduino/tree/rework-tinyusb-lib

Here is the example https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/major-v1/examples/CDC/no_serial/no_serial.ino

**Comment by amowry** [comment]
Thanks so much for this addition!
